src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt

```diff
@@ -752,15 +744,17 @@ class ActivityResourceIntTest : IntegrationTestBase() {
 
       @Test
       fun `should return OK if pay rate removed which is NO LONGER allocated to an offender`() {
-        val offender =
-          repository.save(OffenderBuilder(nomsId = "A1234TT").withBooking(OffenderBookingBuilder(agencyLocationId = PRISON_ID, incentives = listOf(IncentiveBuilder(iepLevel = "STD")))))
-        repository.save(
-          offenderProgramProfileBuilderFactory.builder(
-            payBands = listOf(offenderProgramProfilePayBandBuilderFactory.builder(endDate = LocalDate.now().minusDays(1).toString())),
-          ),
-          offender.latestBooking(),
-          courseActivity,
-        )
+        testData(repository) {
```

Comment:
Hmm - should we not write it like this ?

`
       testData(repository) {
          offender = offender(nomsId = "A1234TT") {
            booking(agencyLocationId = "LEI") {
              incentive(iepLevelCode = "STD")
              courseAllocation(courseActivity) {
                payBand(endDate = yesterday.toString())
              }
            }
          }
        }`

Reply:
As per Slack conv. courseAllocation could live under either an offender booking or under a course activity - and as you rightly pointed out why not allow both? Coming soon.